File: Python/Problem010.py
from math import sqrt; from itertools import count, islice
from time import time

''' 
Problem 10: https://projecteuler.net/problem=10

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
'''

def primeSieve(n):
    sieve = [True] * n
    for i in range(3, int(n**0.5) + 1, 2):
        if sieve[i]:
            sieve[i*i : : 2*i] = [False] * ((n - i*i - 1) // (2 * i) + 1)
    return [2] + [i for i in range(3, n, 2) if sieve[i]]

start = time()
ans = sum(primeSieve(2 * 10**6))

if __name__ == '__main__':
    start = time()
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")

File: CPP/Problem010.cpp
#include "iostream"
#include "vector"
#include "boost/range/irange.hpp"


long Problem010();
std::vector<int> primeSieve(int n);

int main(){
    clock_t start; 
    start = std::clock();
    int ans = Problem010();
    double time_taken = double(clock() - start) / double(CLOCKS_PER_SEC); 

    printf("\nAnswer: %d\n", ans);
    printf("Time Taken: %f seconds\n\n", time_taken);

    return 0;
}

/*
Problem010 answers the problem at : https://projecteuler.net/problem=10

* Problem 10:
		Find the sum of all the primes below two million.
*/
long Problem010() {
    std::vector<int> primes = primeSieve(20);
    long res = 0;

    for (int i: primes){
        printf("%d, ", i);
        res += long(i);
    }

    return res;
}

std::vector<int> primeSieve(int n){
    std::vector<int> pr;
    boost::integer_range<int> nums = boost::irange(1, n);
        

    return pr;
}


